modules/nlu/src/backend/typings.d.ts

```diff
@@ -0,0 +1,54 @@
+declare type EngineByBot = { [botId: string]: Engine }
+
+declare interface Engine {
+  sync(): Promise<void>
+  checkSyncNeeded(): Promise<boolean>
+  extract(event): Promise<Predictions.ExtractResult>
+}
+
+declare interface EntityExtractor {
+  extract(input: string, lang: string): Promise<Predictions.Entity[]>
+}
+
+declare interface IntentClassifier {
+  predict(input: string): Promise<Predictions.Intent[]>
+}
+
+declare interface LanguageIdentifier {
+  identify(input: string): Promise<string>
+}
+
+declare namespace Predictions {
+  interface ExtractResult {
+    intent: Intent
+    intents: Intent[]
+    language: string
+    entities: Entity[]
+  }
+
+  interface Intent {
+    name: string
+    confidence: number
```

Comment:
This is where we should add the helper functions definitions that you add in modules/nlu/src/backend/middleware.ts